nitro-protocol/test/contracts/LedgerFinancingApp/requireStateSupported.test.ts

```diff
@@ -0,0 +1,291 @@
+    //  - the chain is advanced by 1 day
+    //  - challenge outcome is 506:495. Fraud!
+
+    const currentBlockNumber = await provider.getBlockNumber();
+    const supportproofAppData = {
+      ...baseAppData,
+      blocknumber: currentBlockNumber,
+    };
+
+    const supportproofState: State = {
+      ...baseState,
+      appData: appDataABIEncode(supportproofAppData),
+    };
+    const pfStateSignedByBoth: RecoveredVariablePart = {
+      variablePart: getVariablePart(supportproofState),
+      signedBy: BigNumber.from(0b11).toHexString(),
+    };
+
+    advanceOneDay();
+
+    const challengeState: State = {
+      ...baseState,
+      appData: appDataABIEncode(supportproofAppData),
+      turnNum: baseState.turnNum + 1,
+      outcome: computeOutcome({
+        [MAGIC_NATIVE_ASSET_ADDRESS]: {[intermediary]: 506, [merchant]: 494}, // 506 is unfair: should be 505
+      }),
+    };
+    const updatedWithIntermediarySignature: RecoveredVariablePart = {
+      variablePart: getVariablePart(challengeState),
+      signedBy: BigNumber.from(0b01).toHexString(),
+    };
+    await expectRevert(
+      () =>
+        ledgerFinancingApp.requireStateSupported(
+          fixedPart,
+          [pfStateSignedByBoth],
+          updatedWithIntermediarySignature
+        ),
+      'earned<claimed'
+    );
+  });
+
+  it('rejects unilateral unsupported candidates', async () => {
+    // construct challenges with unsupported candidates signed by:
+    // - only the intermediary
+    // - only the merchant
+    // assert failure
+
+    await expectRevert(
+      () => ledgerFinancingApp.requireStateSupported(fixedPart, [], signedByMerchant),
+      '!unanimous; |proof|=0'
+    );
+    await expectRevert(
+      () => ledgerFinancingApp.requireStateSupported(fixedPart, [], signedByIntermediary),
+      '!unanimous; |proof|=0'
+    );
+  });
+
+  it('rejects unilateral support proof states', async () => {
+    // construct challenges where the proof state is signed by
+    //  - only the intermediary
+    //  - only the merchant
+    // assert failure.
+
+    await expectRevert(
+      () =>
+        ledgerFinancingApp.requireStateSupported(
+          fixedPart,
+          [signedByMerchant],
+          signedByIntermediary
+        ),
+      '!unanimous proof state'
+    );
+
+    await expectRevert(
+      () =>
+        ledgerFinancingApp.requireStateSupported(
+          fixedPart,
+          [signedByIntermediary],
+          signedByMerchant
+        ),
+      '!unanimous proof state'
+    );
+  });
+
+  it('rejects too-long proofs', async () => {
+    // construct a challenge with two proof states, assert failure.
+
+    await expectRevert(
+      () =>
+        ledgerFinancingApp.requireStateSupported(
+          fixedPart,
+          [signedByMerchant, signedByIntermediary], // two proof states - should fail
+          signedByIntermediary
+        ),
+      '|proof| > 1'
+    );
+  });
+});
+
+/**
+ * increase blocknumber on provider by 7500.
```

Comment:
Nit: 7500 blocks is 25 hours. So this method could have a more accurate name.

Reply:
Alternatively I could just move it to do exactly 7200 and check that things work. 7500 was picked arbitrarily to avoid any off-by-one time accounting errors.

Reply:
This change accidentally snuck in with 76f8f179, and it worked.